### PR TITLE
Info dump 29-31/66: Minimap and stepping on stuff

### PR DESCRIPTION
**Pull request by End45**
Adds functions related to:

- Minimap data and display
- What happens when a monster steps on an item or a trap

Everything I had for the "RNG" category was already documented, so nothing new there.

### CODE REVIEW

File: headers/functions/overlay29.h [hunk 1]
```diff
@@ -18,10 +18,13 @@ bool FixedRoomIsSubstituteRoom(void);
 bool StoryRestrictionsEnabled(void);
 int GetScenarioBalanceVeneer(void);
 void FadeToBlack(void);
+struct trap* GetTrapData(struct entity* trap_entity);
```
Comment (UsernameFodder) [inline headers/functions/overlay29.h:21]: Do `GetTrapData` and `GetItemData` just return the `info` field? Can we call these `GetTrapInfo` and `GetItemInfo` if so?

File: symbols/overlay29.yml [hunk 2]
```diff
@@ -1085,6 +1134,33 @@ overlay29:
...
+        r2: ?
+        r3: ?
+    - name: ApplyTrapEffect
+      address:
+        EU: 0x22EFB08
+        NA: 0x22EF154
+      description: |-
+        Performs the effect of a triggered trap.
+        
+        The trap's animaton happens before this function is called.
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:1157]: animation

File: symbols/overlay29.yml [hunk 3]
```diff
@@ -2208,6 +2294,16 @@ overlay29:
         
         r0: Pointer to an entity. Can be null.
         returns: Seems to be true if there were any pending actions to display.
+    - name: CheckNonLeaderTile
+      address:
+        EU: 0x23060C0
+        NA: 0x2305694
+      description: |-
+        Similar to CheckLeaderTile, but for other pokémon.
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:2302]: "pokémon" -> "monsters" for consistency?

File: headers/functions/overlay29.h [hunk 4]
```diff
@@ -432,11 +445,17 @@ void SetShouldBoostHiddenStairsSpawnChance(bool value);
...
+void HiddenStairsTrigger(bool show_message);
 undefined4 GetDungeonGenInfoUnk0C(void);
 struct minimap_display_data* GetMinimapData(void);
+void DrawMinimapTile(int x, int y);
+void UpdateMinimap(void);
 void SetMinimapDataE447(uint8_t value);
 uint8_t GetMinimapDataE447(void);
 void SetMinimapDataE448(uint8_t value);
+void InitWeirdMinimapMatrix(void);
+void InitMinimapDisplayTile(struct minimap_display_tile* minimap_display_tile, void* ptr);
```
Comment (UsernameFodder) [inline headers/functions/overlay29.h:458]: I generally reserve `void*` when we _know_ the pointer is generic. If it's a pointer to an unknown struct, we should use `undefined*` instead.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/functions/overlay29.h (hunk 1 above)
Comment (End45) [reply headers/functions/overlay29.h:21]: Sure

**Comment by End45** [comment]
(That last commit isn't a full merge, just a confict fix)

**Review by UsernameFodder — APPROVED** [review]
LGTM, thanks!

**Pull request merged**